test/oracles/adapters/balancer/BalancerV2PriceOracle.test.ts

```diff
@@ -0,0 +1,194 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signers";
+import { expect } from "chai";
+import hre from "hardhat";
+import axios from "axios";
+
+import { EXODIA_ROLES_DID } from "../../../../deploy/38_deployExodiaRoles";
+import { BALANCER_V2_PRICE_ORACLE_DID } from "../../../../deploy/41_deployBalancerV2PriceOracle";
+import { IExodiaContractsRegistry } from "../../../../src/contracts/exodiaContracts";
+import { IExtendedHRE } from "../../../../src/HardhatRegistryExtension/ExtendedHRE";
+import { ZERO_ADDRESS } from "../../../../src/utils";
+import {
+    BalancerV2PriceOracle,
+    BalancerV2PriceOracle__factory,
+    ExodiaRoles,
+    ExodiaRoles__factory,
+} from "../../../../typechain";
+
+const xhre = hre as IExtendedHRE<IExodiaContractsRegistry>;
+const { deployments, get, getNamedAccounts, deploy, getNetwork } = xhre;
+
+const BALANCER_V2_VAULT = "0x20dd72Ed959b6147912C2e529F0a0C651c33c9ce";
+const BEETS = "0xF24Bcf4d1e507740041C9cFd2DddB29585aDCe1e";
+const WFTM = "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83";
+const WBTC = "0x321162Cd933E2Be498Cd2267a90534A804051b11";
+const BEETS_FTM_POOL = "0xcdE5a11a4ACB4eE4c805352Cec57E236bdBC3837";
+const MINIMUM_UPDATE_INTERVAL = 60 * 5; // 5 mins
+const FTM_USD_FEED = "0xf4766552D15AE4d256Ad41B6cf2933482B0680dc";
+const USDC_WFTM_BTC_ETH_POOL = "0xf3A602d30dcB723A74a0198313a7551FEacA7DAc";
```

Comment:
There is an object in src/contracts/index.ts with external address. I think they are mostly all there, you can add the ones missing in the object.

You can use it like this to get the addresses:
```    
const { SPOOKY_ROUTER, EXODDAI_LP } = externalAddressRegistry.forNetwork(
        await getNetwork()
);
```
getNetwork function is available on the HardHatRuntime object, check deployment 27 for reference